udp/client/clientconn.go

```diff
@@ -297,8 +297,10 @@ func (cc *ClientConn) WriteMessage(req *pool.Message) error {
 		req := bwreq.(*pool.Message)
 		if req.Options().HasOption(message.Block1) || req.Options().HasOption(message.Block2) {
 			req.SetMessageID(cc.getMID())
```

Comment:
Shouldn't now the return value of getMID be checked, since it now returns int32? This could create a message with invalid ID if somebody uses a bad ID generator.

Reply:
This is a user responsibility, but if user set invalid message then coder returns error during encoding:
https://github.com/plgd-dev/go-coap/blob/4e633a457ee7c1772e8b78c191a7b3c594ae9755/udp/coder/coder.go#L50